### DISCUSSION TITLE
Apply filters to an expanded field. (like "include" in Prisma)

### DISCUSSION THREAD

**Discussion by jiu-virgil** [pocketbase/pocketbase#4495] [Q&A]
i have a collection subject
i have another collection, categories, that has a relation field to subject.
i have the following query
```js
const categories = await pb.collection('Categories').getList(1, 1, { filter: `subject.title==${params.subject}`, expand: 'subject_id' });
```
it's supposed to check if the subject title i have in my code is the same as the subject in the category's subject title

but it doesn't work. what am i doing wrong?
can you not do expand and then filter by something expanded? didn't find this case in the docs.

**Comment by ganigeorgiev** [comment]
1. The equal operator should be `=`, not `==`. For list with all available operators, please refer to https://pocketbase.io/docs/api-rules-and-filters/.

2. String literals must be single or double quoted (aka. wrap `${params.subject}` in quotes). Alternatively, you can also use the [`pb.filter()` helper](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#binding-filter-parameters).

**Reply by jiu-virgil** [reply]
It's true, I've made those errors fiddling around with the functions trying to get something working.
But my question still stands. 

For example, if i have a `Posts` collection and an `Authors` collection, the `Posts` collection having a relation filed of `author_id` and I want to query the Posts collection and filter by an `author.name` instead of `author.id`, 
do I have to make a query with all `Posts` and then filter the result in javascript to only get that `posts.items.expand.author.name` I'm interested in? I'm guessing this approach works but isn't the best.

If it helps, I know in Prisma this would simply be
```js
const posts = await prisma.post.findMany({
  where: {
    author: {
      name: authorName,
    },
  },
  include: {
    author: true, // Optionally include author information in the result
  },
});
```

**Reply by ganigeorgiev** [reply]
I'm confused and not sure that fully understand the desired behavior.

In general if you want to filter by a relation field you can simple use dot-notation to target the specific field. For example:

```js
const name = "something"

const records = await pb.collection("Posts").getFullList({
    filter: pb.filter("author_id.name = {:name}", { name }),
    expand: "author_id",
})
```

Note that the expand will work only if your client request satisfy the View API rule of the expanded collection (in the above example the "Authors").

For more details about the relations you can refer to https://pocketbase.io/docs/working-with-relations/.
